sql-plugin/src/main/330+/scala/org/apache/spark/sql/rapids/shims/intervalExpressions.scala

```diff
@@ -523,6 +524,11 @@ case class GpuDivideDTInterval(
   }
 
   override def doColumnar(interval: GpuColumnVector, num: GpuColumnVector): ColumnVector = {
+    withResource(makeZeroScalar(num.getBase.getType)) { zeroScalar =>
+      if (num.getBase.contains(zeroScalar)) {
+        throw RapidsErrorUtils.divByZeroError(origin)
```

Comment:
If we use a new error `SparkArithmeticException`, it's better to remove the original error `ArithmeticException`.
Can we just update the original check?

[Original check](https://github.com/NVIDIA/spark-rapids/blob/branch-22.08/sql-plugin/src/main/330%2B/scala/org/apache/spark/sql/rapids/shims/intervalExpressions.scala#L244toL247)
```
    if (IntervalUtils.hasZero(q)) {
      throw new ArithmeticException("overflow: interval / zero")
    }
```

Original error:
```
java.lang.ArithmeticException: overflow: interval / zero
	at org.apache.spark.sql.rapids.shims.IntervalUtils$.divWithHalfUpModeWithOverflowCheck(intervalExpressions.scala:246)
```

Reply:
> Should also check another path, the `numScalar` can be zero.
> 
> ```
> override def doColumnar(interval: GpuColumnVector, numScalar: GpuScalar)
> ```

Updated!

Reply:
> If we use a new error `SparkArithmeticException`, it's better to remove the original error `ArithmeticException`. Can we just update the original check?
> 
> [Original check](https://github.com/NVIDIA/spark-rapids/blob/branch-22.08/sql-plugin/src/main/330%2B/scala/org/apache/spark/sql/rapids/shims/intervalExpressions.scala#L244toL247)
> 
> ```
>     if (IntervalUtils.hasZero(q)) {
>       throw new ArithmeticException("overflow: interval / zero")
>     }
> ```
> 
> Original error:
> 
> ```
> java.lang.ArithmeticException: overflow: interval / zero
> 	at org.apache.spark.sql.rapids.shims.IntervalUtils$.divWithHalfUpModeWithOverflowCheck(intervalExpressions.scala:246)
> ```

Updated!.

BTW, I don't update this error:
```scala
  def divWithHalfUpModeWithOverflowCheck(p: BinaryOperable, q: BinaryOperable): ColumnVector = {
    // 1. overflow check q is 0
    if (IntervalUtils.hasZero(q)) {
      throw new ArithmeticException("overflow: interval / zero")
    }
    ...
```
because we don't have the value `origin: Origin` in this context